state/executor.go

```diff
@@ -303,9 +303,9 @@ func (e *BlockExecutor) execute(ctx context.Context, state State, block *types.B
 	}
 
 	for _, tx := range block.Data.Txs {
-		res := e.proxyApp.DeliverTxAsync(abci.RequestDeliverTx{Tx: tx})
-		if res.GetException() != nil {
-			return nil, errors.New(res.GetException().GetError())
+		_ = e.proxyApp.DeliverTxAsync(abci.RequestDeliverTx{Tx: tx})
+		if e.proxyApp.Error() != nil {
+			return nil, err
```

Comment:
Are you sure this is right? you're returning the `err` from https://github.com/celestiaorg/optimint/blob/935c60a3e1ab2e13c34bf4cdca0dd6ce37e32e1f/state/executor.go#L291 here.